2023-09-20-issue1-seed-identification-api-for-nachet-frontend.py

```diff
@@ -0,0 +1,150 @@
+    
+    query = (
+    f"SELECT regexp_replace("
+    f"'{seed_full_url}', '.*seeds-identification/([^/]+).*', '\\1') AS seed_name;"
+    )
+    cursor.execute(query)
+    seed_name_query = cursor.fetchall()
+    
+    if seed_name_query:
+        seed_name = seed_name_query[0]['seed_name']
+        url_to_seed_mapping[seed_full_url] = seed_name
+
+# Now you have a dictionary where the keys are URLs and the values are seed names
+print("\nList of selected seeds :")
+for url, seed_name in url_to_seed_mapping.items():
+    print(f"{seed_name}")
+
+def seed_identification_api(system_prompt, user_prompt, json_template):
+    """
+    Process seed data using Azure OpenAI endpoint and save results as JSON files.
+
+    Args:
+        system_prompt (str): A system prompt for the OpenAI conversation.
+        user_prompt (str): A user prompt for the OpenAI conversation.
+        json_template (json): A JSON template for the OpenAI request.
+
+    This function performs the following steps:
+    1. Iterates through a list of seed values.
+    2. Checks if a JSON file for each seed exists and skips if it does.
+    3. Constructs an SQL query to retrieve data related to the seed from a database.
+    4. Sends the query to the database and fetches the retrieved data.
+    5. Concatenates the cleaned content into a single 'page.'
+    6. Sends a request to the Azure OpenAI endpoint to get a response.
+    7. Processes the response, extracting the scientific name and saving it as a JSON file.
+    """
+    for url, seed_name in url_to_seed_mapping.items():
+        print("\nCurrent seed : " + seed_name)
+        seed_json_path = seed_name + ".json"
+
+        if nachet.json_file_exists(seed_data_path, seed_json_path):
+            print(f"The JSON file {seed_json_path} exists in {seed_data_path}, skipping")
+        else:
+            query = """
+            SELECT
+                hc.md5hash,
+                REGEXP_REPLACE(hc.content, '<[^>]+>', '', 'g')
+            AS
+                cleaned_content
+            FROM
+                html_content hc
+            INNER JOIN
+                crawl c ON hc.md5hash = c.md5hash
+            WHERE
+                 c.url = '""" + url + """';
+            """
+            cursor.execute(query)
+            web_pages_fr_en = cursor.fetchall()
+
+            all_language_seed_page = ""
+            for row in web_pages_fr_en:
+                web_text = row.get('cleaned_content')
+                all_language_seed_page += web_text
+            page = all_language_seed_page
+            md5hash = row.get('md5hash') 
+
+            ### Get the images corresponding to the current page
+            query = """
+            SELECT DISTINCT
+                image_links[1] AS photo_link,
+                image_descriptions[1] AS photo_description
+            FROM (
+                SELECT
+                    (regexp_matches(content, 'src="([^"]+)"', 'g')) AS image_links,
```

Comment:
Je pense qut tu devrais peut etre commenter pourquoi tu fais des regex comme ça

Reply:
La fonction regex_matches en SQL permet d'extraire des morceaux de texte d'une colonne en fonction de règles définies par des expressions. En d'autres termes, elle te permet d'obtenir des parties spécifiques de la donnée qui correspondent à un modèle de texte que tu définis à l'aide de l'expression régulière. Par exemple, ici, on extrait toutes les images d'une page en utilisant regex_matches avec un modèle qui correspond aux balises des images.

Reply:
ok got it, merci :)